sdk/python/lib/setup.py

```diff
@@ -45,7 +45,7 @@ def readme():
       # Keep this list in sync with Pipfile
       install_requires=[
           'protobuf~=4.21',
-          'grpcio==1.47',
+          'grpcio==1.50',
```

Comment:
QQ: Does this need to be `==` pinned? `>=` should work, I think, and `1.51` is already released. 

Example of `>=` pinning: https://github.com/googleapis/python-api-core/pull/466/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R42.

(Great to see this change happening)

Reply:
It would be `~=` like the other dependencies here, but I think we're a bit less keen on that for grpcio because we've had multiple cases of them releasing bad versions and all our pipelines breaking because of it.

Reply:
Note that version 1.51 has already been yanked, so likely would have caused issues.